be/src/storage/lake/tablet_manager.cpp

```diff
@@ -181,18 +181,19 @@ Status TabletManager::create_tablet(const TCreateTabletReq& req) {
                 col_idx_to_unique_id[i] = next_unique_id++;
             }
         }
-        RETURN_IF_ERROR(starrocks::convert_t_schema_to_pb_schema(
-                mutable_new_schema, next_unique_id, col_idx_to_unique_id, tablet_metadata_pb.mutable_schema()));
+        RETURN_IF_ERROR(
+                starrocks::convert_t_schema_to_pb_schema(mutable_new_schema, next_unique_id, col_idx_to_unique_id,
+                                                         tablet_metadata_pb.mutable_schema(), req.compression_type));
```

Comment:
You should check whether compression_type is set in the `req`.
because this field is not set in the request from old version FE,

Reply:
The check of req and the error return of switch case in convert_t_schema_to_pb_schema can be considered as the same thing, and I will fix it together.